src/main/java/tc/oc/pgm/restart/RestartManager.java

```diff
@@ -94,6 +97,23 @@ public boolean isRestartDeferredBy(RequestRestartEvent.Deferral deferral) {
     return this.deferrals.contains(deferral);
   }
 
+  public void requestTimedRestart(Duration duration) {
```

Comment:
What happens if this method was already called? will the previous task get cancelled? Doesn't seem to do that right now.

Reply:
Ah true, fixed.